core/src/nucleus/ribosome/api/commit.rs

```diff
@@ -83,29 +79,26 @@ pub fn invoke_commit_app_entry(
     );
 
     let maybe_json = match task_result {
-        Ok(action_response) => match action_response {
-            ActionResponse::Commit(_) => action_response.to_json(),
-            _ => return ribosome_error_code!(ReceivedWrongActionResult),
-        },
+        Ok(address) => serde_json::to_string(&CommitEntryResult::success(address)),
+        Err(HolochainError::ValidationFailed(fail_string)) => {
+            serde_json::to_string(&CommitEntryResult::failure(fail_string))
```

Comment:
So... We didn't have a way before to pass back the validation failure reason?

Reply:
Yes, we did, but it was handled exactly like an error below. The tests in HDK and app-spec were tailored for that. This is cleaner and more robust.